src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SELECT_BOOK_BY_ID)) {
+            statement.setLong(ID_COLUMN_INDEX, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(createNewBook(
+                        id,
+                        resultSet.getString("title"),
+                        resultSet.getBigDecimal("price")
+                ));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Не вдалося знайти книгу за id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SELECT_ALL_BOOKS)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                books.add(createNewBook(
+                        resultSet.getLong("id"),
+                        resultSet.getString("title"),
+                        resultSet.getBigDecimal("price")
+                ));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Не вдалося знайти всі книги", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_BOOK)) {
+            statement.setString(TITLE_COLUMN_INDEX, book.getTitle());
+            statement.setBigDecimal(PRICE_COLUMN_INDEX, book.getPrice());
+            statement.setLong(ID_COLUMN_INDEX + 1, book.getId());
+            if (statement.executeUpdate() < 1) {
+                throw new DataProcessingException("Не вдалося оновити книгу: " + book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Не вдалося оновити книгу: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE_BOOK_BY_ID)) {
+            statement.setLong(ID_COLUMN_INDEX, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Не вдалося видалити книгу з id: " + id, e);
+        }
+    }
+
+    private Book createNewBook(Long id, String title, BigDecimal price) {
```

Comment:
it's better to provide ResultSet resultSet
```suggestion
    private Book mapResultSetToBook(ResultSet resultSet) {
```